src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorJsonDataPayload.java

```diff
@@ -0,0 +1,528 @@
+       * @since 1.0.0
+       */
+      public synchronized Object getValue() {
+        return value;
+      }
+
+      /**
+       * Gets the unit of the series (associated with the value).
+       *
+       * @return the unit of the series (associated with the value)
+       * @since 1.0.0
+       */
+      public synchronized String getUnit() {
+        return unit;
+      }
+
+      /**
+       * Gets the original time of the series data.
+       *
+       * @return the original time of the series data
+       * @since 1.0.0
+       */
+      public synchronized Date getOriginalTime() {
+        return originalTime;
+      }
+    }
+
+    /**
+     * Gets the {@link Map} object representing the series of data for the fragment. This is the map
+     * of series names to the list of associated series data.
+     *
+     * <p>The type-parameterized version of this map would be {@code Map<String,List<Series>>}.
+     *
+     * @return the {@link Map} object representing the series of data for the fragment
+     * @since 1.0.0
+     */
+    public synchronized Map getSeriesMap() {
+      return seriesMap;
+    }
+
+    /**
+     * Adds the specified series to the fragment with the specified name. If there is already a
+     * series with the specified name, the series is updated with the new series data.
+     *
+     * @param seriesName the name of the series
+     * @param series the series to add
+     * @since 1.0.0
+     */
+    public synchronized void addSeries(String seriesName, Series series) {
+      if (seriesMap.containsKey(seriesName)) {
+        List existingSeriesList = (List) seriesMap.get(seriesName); // List<Series>
+        existingSeriesList.add(series);
+      } else {
+        List seriesList = new ArrayList(); // List<Series>
+        seriesList.add(series);
+        seriesMap.put(seriesName, seriesList);
+      }
+    }
+  }
+
+  /**
+   * Constructor for a new {@link CConnectorJsonDataPayload} object with the specified date and type
+   * (child device).
+   *
+   * @param time the {@link Date} object representing the timestamp of the data payload. This is the
+   *     timestamp associated with the aggregated data in the payload.
+   * @param type the {@link String} object representing the type of the data payload. This usually
+   *     is the name of the child device the data is associated with, or {@link #TYPE_NONE} if the
+   *     data is not associated with a child device. If {@code null} is specified, {@link
+   *     #TYPE_NONE} is used.
+   * @throws JSONException if unable to populate the external source information object when the
+   *     specified {@code type} value is non-null
+   * @since 1.0.0
+   */
+  public CConnectorJsonDataPayload(Date time, String type) throws JSONException {
+    this.time = time;
+    this.type = type != null ? type : TYPE_NONE;
+
+    // Populate external source field if the type is not null/none, otherwise set to null
+    if (type != null) {
```

Comment:
The comment above indicates if the type is not null/none, but the if only checks for null
```java
   if ( !(type == null || type == TYPE_NONE) ) {
```
caution that spacing is not correct, also not %100 confident I know java logical or

Reply:
Updated to instead reference `this.type`, which is checked/set above.